HL-560/570: Hide expired tet postings from search

nenonja opened the pull request:
## Description :sparkles:

- Hide expired test postings from tet-search
- Expired postings section on tet-admin
- hack: Use special end date value `2999-12-31`to express never-expiring posting

Commits are self-explanatory and in logical parts:

1. [fix: posting page does not work](https://github.com/City-of-Helsinki/yjdh/pull/1564/commits/6786a48a9c79952727d4f586106d5801e9a5d46f)
2. [fix: tet youth tests fail because passWithNoTests flag is not set](https://github.com/City-of-Helsinki/yjdh/pull/1564/commits/da0845519ff64760d6d6c59c0ff370b8bb629ced)
4. [refactor: simplify eventsToTetPostings function](https://github.com/City-of-Helsinki/yjdh/pull/1564/commits/f089448d32b6c61a9bd1b055cf503da4f1539628)
5. [feat(hl-570): tet-admin: create different section for expired postings](https://github.com/City-of-Helsinki/yjdh/pull/1564/commits/5c268754c2ed8bf12238102c10964111eb9b9503)
6. [feat(hl-570): Hack: use special end date '31.12.2999' to indicate nev…](https://github.com/City-of-Helsinki/yjdh/pull/1564/commits/53aa30f0df37e3d4b22e1108cac4c19ba6e6f456) 
7. [fix(hl-560): hide ended events from tet list](https://github.com/City-of-Helsinki/yjdh/pull/1564/commits/840eee97e0d169bae237a040d7de425b0ad3b00f)
8. [chore: make index page test more robust by waiting for loading to com…](https://github.com/City-of-Helsinki/yjdh/pull/1564/commits/6b6d62bc1a052691305fb5e3a4feb7267b14e9f4) 
9. [chore: better debugging for exceptions in comment](https://github.com/City-of-Helsinki/yjdh/pull/1564/commits/08092e40dc7b73e38e62ad7c7d82284184494430)
10. [refactor: remove duplicate date util methods from tet](https://github.com/City-of-Helsinki/yjdh/pull/1564/commits/cefe01f6d2a3c96bd04bf46ff09bc51c9ec1339c)

## Issues :bug:

## Testing :alembic:

## Screenshots :camera_flash:

## Additional notes :spiral_notepad:

karisal-anders commented on frontend/tet/shared/src/utils/get-server-end-date.ts:
```diff
@@ -0,0 +1,10 @@
+import { furthestServerEndDate } from 'tet-shared/constants/furthest-end-date';
+
+export default (dateString?: string): string => {
+  // if end date is undetermined, return furthest end dare special value
```
typo dare→date

karisal-anders commented on frontend/tet/admin/src/components/jobPostings/__tests__/JobPostingsListItem.test.tsx:
```diff
@@ -61,10 +66,20 @@ describe('JobPostingsListItem', () => {
+
+  it('it should hide end date when furthest end date special value used', async () => {
+    renderComponent(<JobPostingsListItem posting={never_expiring} />);
+    await screen.findByText(new RegExp(`${never_expiring.start_date}-$`, 'i'), {
```
If I understand the regular expression correctly this only tests that "<start_date>-" is found in
the string but it doesn't say anything about what comes or doesn't come after it.

nenonja replied:
Actually the `$` in the end checks that there is nothing coming after that.

karisal-anders replied:
Ah, so it's testing that a string ends, next is "<start_date>-" and that string ends. Hmm... so there's a row with only "<start_date>-" in the text, right?

nenonja replied:
Yes, the idea is to check that when end date is the special value `2999-12-31` then it does not display `<start_date>-2999-12-31` but just `<start_date>-` instead

karisal-anders replied:
Ah, yes, the first $ is related to the variable evaluation i.e. `${x}` evaluates variable x but the last $ is the end of string regular expression marker, alright.

nenonja replied:
For more clarity I changed it to `^${never_expiring.start_date}-$`

nenonja replied:
> Left some comments, please have a look at them.
> 
> backend/tet/events/linkedevents.py: in `list_ongoing_events_authenticated` function this could be also removed if doing changes to the pull request:
> 
> ```python
> # TODO exclude ended events
> # "start": "2021-01-01",
> # "end"
> ```
> 
> as the expired events are not excluded but partitioned into a different group afterwards now.

Removed

karisal-anders commented on backend/tet/events/linkedevents.py:
```diff
@@ -65,11 +66,13 @@ def _api_call(
                 )
             # TODO 403 is actually configuration error in Linked Events for the TET API key
             elif e.response.status_code == 401 or e.response.status_code == 403:
+                # print('error is: ' + e.response.text)
```
I would not include this commit because putting code in comments is normally not
good code practice. If the code should be included then it shouldn't be in comments
and if it shouldn't be included then it should be deleted. There are exceptions as to
almost every case but I don't think this is one of those. What do you think?

nenonja replied:
it can be removed. I just spent at least half an hour to find a way to print the whole error message. `e` or `e.response` didnt print the message. but perhaps "real" backend developers know instantly how to print or debug the error :)

nenonja replied:
This commit is now deleted

karisal-anders reviewed (commented):
Left some comments, please have a look at them.

backend/tet/events/linkedevents.py: in `list_ongoing_events_authenticated` function this could be also removed if doing changes to the pull request:
```python
# TODO exclude ended events
# "start": "2021-01-01",
# "end"
```
as the expired events are not excluded but partitioned into a different group afterwards now.

nenonja commented:
> Left some comments, please have a look at them.
> 
> backend/tet/events/linkedevents.py: in `list_ongoing_events_authenticated` function this could be also removed if doing changes to the pull request:
> 
> ```python
> # TODO exclude ended events
> # "start": "2021-01-01",
> # "end"
> ```
> 
> as the expired events are not excluded but partitioned into a different group afterwards now.

Removed

karisal-anders reviewed (approved):
LGTM
